api/app/src/mappings/google/activity.ts

```diff
@@ -15,30 +31,91 @@ export const mapToActivity = (googleActivity: GoogleActivity, date: string): Act
     metadata: metadata,
   };
 
-  googleActivity.bucket[0].dataset.forEach(data => {
-    if (data.point.length) {
-      const values = getValues(data.point);
-      const intValues = getValues(data.point, ValueKey.intVal);
-
-      if (data.dataSourceId === sourceIdCalories) {
-        activity.summary = {
-          ...activity.summary,
-          energy_expenditure: {
-            active_kcal: sum(values),
-          },
-        };
-      }
+  if (googleActivity) {
+    googleActivity.bucket[0].dataset.forEach(data => {
+      if (data.point.length) {
+        const values = getValues(data.point);
+        const intValues = getValues(data.point, ValueKey.intVal);
+
+        if (data.dataSourceId === sourceIdActiveMinutes) {
+          const seconds = convert(sum(intValues)).from("min").to("s");
+
+          activity.summary = {
+            ...activity.summary,
+            durations: {
+              ...activity.summary?.durations,
+              active_seconds: seconds,
+            },
+          };
+        }
+
+        if (data.dataSourceId === sourceIdCalories) {
+          activity.summary = {
+            ...activity.summary,
+            energy_expenditure: {
+              active_kcal: sum(values),
+            },
+          };
+        }
+
+        if (data.dataSourceId === sourceIdSteps) {
+          activity.summary = {
+            ...activity.summary,
+            movement: {
+              ...activity.summary?.movement,
+              steps_count: sum(intValues),
+            },
+          };
+        }
+
+        if (data.dataSourceId === sourceIdDistance) {
+          activity.summary = {
+            ...activity.summary,
+            movement: {
+              ...activity.summary?.movement,
+              distance_meters: sum(values),
+            },
+          };
+        }
 
-      if (data.dataSourceId === sourceIdSteps) {
-        activity.summary = {
-          ...activity.summary,
-          movement: {
-            steps_count: sum(intValues),
-          },
-        };
+        if (data.dataSourceId === sourceIdSpeed) {
+          const convertSpeed = convert(sum(values)).from("m/s").to("km/h");
+          const { max_item } = Util.getMinMaxItem(values);
+
+          activity.summary = {
+            ...activity.summary,
+            movement: {
+              ...activity.summary?.movement,
+              speed: {
+                ...activity.summary?.movement?.speed,
+                avg_km_h: convertSpeed,
+                max_km_h: convert(max_item).from("m/s").to("km/h"),
+              },
+            },
+          };
+        }
       }
-    }
-  });
+    });
+  }
+
+  if (googleSessions) {
+    const sleepType = 72;
```

Comment:
putting this in a shared const would be good since we're also using it in `api/app/src/providers/google.ts`